Tools/Scripts/webkitpy/w3c/test_importer.py

```diff
@@ -212,6 +212,14 @@ def __init__(self, port, test_paths, options):
     def do_import(self):
         if self.source_directory:
             source_path = str(Path(self.source_directory) / 'web-platform-tests')
+            if not Path(source_path).exists():
+                _log.error("There isn't a web-platform-tests directory in the path you have provided: %s" % source_path)
+                exit(1)
+            for test_path in self.test_paths:
+                full_test_path = Path(self.source_directory) / test_path
+                if not Path(full_test_path).exists():
+                    _log.error("The source directory doesn't exist: %s" % full_test_path)
+                    exit(1)
```

Comment:
This should be merged into the `for test_path in self.test_paths` below, because this is still meaningful when `import-w3c-tests` has itself fetched the repo.

Also similar comments to the above appear; this should be using `self.filesystem` and our filesystem abstraction rather than Python's native filesystem stuff, because otherwise it won't go via our mock and hence will fail when we're using our mocked filesystem; we also want it to be a directory and not just any file.

Reply:
Yes but it doesn't have the same exact meaning.

* `import-w3c-tests web-platform-tests/digital-invalid` will fetch remote WPT and modify import json. It can't really verify it exists. 
* `import-w3c-tests web-platform-tests/digital-valid` will fetch remote WPT and create the new hierarchy inside OpenSource and adjust the json. (NO ISSUE)
* `import-w3c-tests web-platform-tests/bogus --src-dir ~/dev` (where `~/dev/web-platform-tests/bogus` does NOT exist). This is what it is about. Don't do anything, just quit. There is no work todo, you local repo has not the right information and fail at the first occurence. 

The `for test_path in self.test_paths` below is indeed doing that in both cases, but return, instead of stopping. So that would not really change the behavior and there is no benefit. 

But if we can't exit at all your previous comment. I'm not sure the full PR makes sense. 

Let's see what @marcoscaceres says as it was a enhancement request by him.

Thanks a lot @gsnedders for the detailed review and explanations.